src/Redux/actionCreators/UserActions.js

```diff
@@ -129,3 +133,97 @@ export const checkWalletStatus = (dispatch, getState) => {
       }
     });
 };
+
+const userDeleted = ({ history, route }) => dispatch => {
+  history.push(route);
+  dispatch({
+    type: SET_USER_DETAILS,
+    payload: {
+      login: {
+        isLoggedIn: false,
+      },
+      isEmailVerified: false,
+      isWalletAssigned: false,
+      email: "",
+    },
+  });
+};
+const deleteUser = user =>
+  new Promise((resolve, reject) => {
+    user.deleteUser(error => {
+      if (error) {
+        reject(error);
+      }
+      resolve();
+    });
+  });
+
+const fetchCurrentUser = () => {
+  return Auth.currentAuthenticatedUser({ bypassCache: true });
+};
+
+export const deleteUserAccount = ({ history, route }) => dispatch => {
+  dispatch(() =>
+    fetchCurrentUser().then(user => {
+      dispatch(() =>
+        deleteUser(user).then(() => {
+          dispatch(userDeleted({ history, route }));
+        })
+      );
+    })
+  );
+};
+
+const forgotPasswordInit = dispatch => {
+  dispatch(loaderActions.startAppLoader);
+  dispatch(errorActions.resetForgotPasswordError);
+};
+
+const forgotPasswordSuccessfull = ({ username, history, route }) => dispatch => {
+  dispatch(updateUsername(username));
+  dispatch(loaderActions.stopAppLoader);
+  history.push(route);
+};
+
+const forgotPasswordFailure = error => dispatch => {
+  dispatch(loaderActions.stopAppLoader);
```

Comment:
Stop the app loader all the other states are updated. This way you will have all the state needed before you remove the loader.

Reply:
[done](https://github.com/singnet/refactored-dapp/pull/17/commits/e7ea7f0a4fec748b8c99056ea84feefadcd02759)